Compiler errors when "theora=yes" caused by memcpy

When I compile with "theora=yes" in order to enable Theora support, I get the following compile error message:

    drivers/theoraplayer/video_stream_theoraplayer.cpp:391:38: error: 'memcpy' was not declared in this scope
    memcpy(ptr, f->getBuffer(), imgsize);
                                       ^

I can workaround this issue on Linux adding the following line:

    #include <string.h>

but I don't know if this could be the solution for all the other platforms.

I think the correct way to solve it would be to use "copymem" and implement
that as memcpy whenever it's available. I was lazy and put memcpy there
because it was a lot faster than the default implementation of copymem.

On Mon, May 4, 2015 at 3:30 PM, Ricardo Pérez <notifications@github.com>
wrote:

> When I compile with "theora=yes" in order to enable Theora support, I get
> the following compile error message:
>
> drivers/theoraplayer/video_stream_theoraplayer.cpp:391:38: error: 'memcpy' was not declared in this scope
> memcpy(ptr, f->getBuffer(), imgsize);
>                                        ^
>
> I can workaround this issue on Linux adding the following line:
>
> #include <string.h>
>
> but I don't know if this could be the solution for all the other platforms.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/issues/1806>.
>

--- a/drivers/theoraplayer/video_stream_theoraplayer.cpp
+++ b/drivers/theoraplayer/video_stream_theoraplayer.cpp
@@ -388,7 +388,7 @@ void VideoStreamTheoraplayer::pop_frame(Ref<ImageTexture> p_tex) {
 	{
 		DVector<uint8_t>::Write wr = data.write();
 		uint8_t* ptr = wr.ptr();
-		memcpy(ptr, f->getBuffer(), imgsize);
+		copymem(ptr, f->getBuffer(), imgsize);
 	}
     /*
     for (int i=0; i<h; i++) {

